In [2]:
import pandas as pd
from ast import literal_eval

In [3]:
movies_metadata = pd.read_csv("data/movies_metadata.csv")
movies_metadata.shape

C:\Users\Admin\AppData\Local\Temp\ipykernel_12704\1855133637.py:1: DtypeWarning: Columns (10) have mixed types. Specify dtype option on import or set low_memory=False.
  movies_metadata = pd.read_csv("data/movies_metadata.csv")


(45466, 24)

In [5]:
movies_metadata.head()
movies_metadata['genres'].head()

0    [{'id': 16, 'name': 'Animation'}, {'id': 35, '...
1    [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...
2    [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...
3    [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...
4                       [{'id': 35, 'name': 'Comedy'}]
Name: genres, dtype: object

In [6]:
print(movies_metadata['genres'][0])

[{'id': 16, 'name': 'Animation'}, {'id': 35, 'name': 'Comedy'}, {'id': 10751, 'name': 'Family'}]


In [7]:
movies_metadata.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45466 entries, 0 to 45465
Data columns (total 24 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   adult                  45466 non-null  object 
 1   belongs_to_collection  4494 non-null   object 
 2   budget                 45466 non-null  object 
 3   genres                 45466 non-null  object 
 4   homepage               7782 non-null   object 
 5   id                     45466 non-null  object 
 6   imdb_id                45449 non-null  object 
 7   original_language      45455 non-null  object 
 8   original_title         45466 non-null  object 
 9   overview               44512 non-null  object 
 10  popularity             45461 non-null  object 
 11  poster_path            45080 non-null  object 
 12  production_companies   45463 non-null  object 
 13  production_countries   45463 non-null  object 
 14  release_date           45379 non-null  object 
 15  re

In [8]:
links_small = pd.read_csv("data/links_small.csv")
links_small.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9125 entries, 0 to 9124
Data columns (total 3 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   movieId  9125 non-null   int64  
 1   imdbId   9125 non-null   int64  
 2   tmdbId   9112 non-null   float64
dtypes: float64(1), int64(2)
memory usage: 214.0 KB


In [9]:
links_small.head()

movieId  imdbId   tmdbId
0        1  114709    862.0
1        2  113497   8844.0
2        3  113228  15602.0
3        4  114885  31357.0
4        5  113041  11862.0

In [ ]:
links_small = links_small[links_small['tmdbId'].notnull()]['tmdbId'].astype('int')
links_small.shape


In [20]:
links_small.info()

<class 'pandas.core.series.Series'>
Index: 9112 entries, 0 to 9124
Series name: tmdbId
Non-Null Count  Dtype
--------------  -----
9112 non-null   int64
dtypes: int64(1)
memory usage: 142.4 KB


In [22]:
movies_metadata.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45466 entries, 0 to 45465
Data columns (total 24 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   adult                  45466 non-null  object 
 1   belongs_to_collection  4494 non-null   object 
 2   budget                 45466 non-null  object 
 3   genres                 45466 non-null  object 
 4   homepage               7782 non-null   object 
 5   id                     45466 non-null  object 
 6   imdb_id                45449 non-null  object 
 7   original_language      45455 non-null  object 
 8   original_title         45466 non-null  object 
 9   overview               44512 non-null  object 
 10  popularity             45461 non-null  object 
 11  poster_path            45080 non-null  object 
 12  production_companies   45463 non-null  object 
 13  production_countries   45463 non-null  object 
 14  release_date           45379 non-null  object 
 15  re

In [24]:
# movies_metadata['id'].isin(links_small.astype('str'))
movies_metadata_small= movies_metadata[movies_metadata['id'].isin(links_small.astype('str'))]
movies_metadata_small.shape

(9099, 24)

In [25]:
# movies = movies_metadata_small['title']
movies = movies_metadata_small[['title','popularity','genres','release_date']]
movies.head()

title popularity  \
0                    Toy Story  21.946943   
1                      Jumanji  17.015539   
2             Grumpier Old Men    11.7129   
3            Waiting to Exhale   3.859495   
4  Father of the Bride Part II   8.387519   

                                              genres release_date  
0  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...   1995-10-30  
1  [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...   1995-12-15  
2  [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...   1995-12-22  
3  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...   1995-12-22  
4                     [{'id': 35, 'name': 'Comedy'}]   1995-02-10

In [33]:
movies.head(1)

title popularity                                             genres  \
0  Toy Story  21.946943  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...   

  release_date  
0   1995-10-30

In [38]:
# print(movies.iloc[0]['genres'])
# print(movies.loc[0,'title'])
print(movies.loc[0,['title','genres']])

title                                             Toy Story
genres    [{'id': 16, 'name': 'Animation'}, {'id': 35, '...
Name: 0, dtype: object


In [42]:
movies['genres'].loc[0]

"[{'id': 16, 'name': 'Animation'}, {'id': 35, 'name': 'Comedy'}, {'id': 10751, 'name': 'Family'}]"

##### [{'id': 16, 'name': 'Animation'}, {'id': 35, 'name': 'Comedy'}, {'id': 10751, 'name': 'Family'}]
##### -> ['Animation','Comedy','Family']
##### -> Animation Comedy Family

In [43]:
df = pd.DataFrame({
    "math":[80,90,100],
    "eng":[75,85,95]
})

In [45]:
df.apply(lambda x:x.mean())

math    90.0
eng     85.0
dtype: float64